tests/unit/cloudformation/test_connection.py

```diff
@@ -149,6 +150,11 @@ def test_update_stack_all_args(self):
             'NotificationARNs.member.2': 'arn:notify2',
             'Parameters.member.1.ParameterKey': 'KeyName',
             'Parameters.member.1.ParameterValue': 'myKeyName',
+            'Parameters.member.2.ParameterKey': 'KeyName2',
+            'Parameters.member.2.ParameterValue': '',
```

Comment:
Thanks! I think we will need to do the update to the logic.

I just tested the integration tests where you set the previous value to true:
https://github.com/boto/boto/pull/3029/files#diff-e70f6fdc16bd22f32c84d3977d1d9f5cR151

It looks like this passes because the value in the second element of the tuple is ``''``. If you put a value like ``'foo'`` in as the second element, you get the following error:
```
boto: ERROR: 400 Bad Request
boto: ERROR: {"Error":{"Code":"ValidationError","Message":"Invalid input for parameter key Parameter1. Cannot specify usePreviousValue as true and non empty value for a parameter","Type":"Sender"},"RequestId":"b6c9aeeb-cdab-11e4-9e3d-df34bd1ee4bf"}
```

Don't worry, the fix is easy to make the ``'foo'`` test case work.

Just make the logic in ``connection.py``:
```
params['Parameters.member.%d.ParameterKey' % (i + 1)] = key
if use_previous:
    params['Parameters.member.%d.UsePreviousValue' 
                 % (i + 1)] = self.encode_bool(use_previous)
else:
    params['Parameters.member.%d.ParameterValue' % (i + 1)] = value

```
I tested the integration test and that worked for me.